### ISSUE TITLE
specimen code not completely detected: including TWO hyphen

### ISSUE THREAD

**Issue by myrmoteras** [gsautter/goldengate-imagine#733]
this specimen code has not been entirely detected, missing the -42
![image](https://user-images.githubusercontent.com/4609956/63674467-e6cb4f00-c7e6-11e9-9130-6a427c9bfa93.png)

**Comment by myrmoteras** [comment]
![image](https://user-images.githubusercontent.com/4609956/63674588-23974600-c7e7-11e9-92be-11e337dddb48.png)

**Comment by gsautter** [comment]
I don't think the hyphen is the problem ... capitalization (or lack thereof) is: both "As" and "Scorp" are not mainly in upper case. Finding any kind of overall regularity in specimen codes looks increasingly hard ... I'll try and think up some solution.

**Comment by gsautter** [comment]
Got it ... the actual problem is the absence of collection codes, as specimen codes are only ever sought right after the former, which also is where they usually occur.

Or from another angle, the problem is of syntactic nature: the collection codes fail to tag because they are hyphen-bound to the specimen codes. Also, the list of collection codes used (Page 5) is intertwined with other content and thus fails to tag properly, so collection code detection fails, and with it, for aforementioned filter, specimen code detection.
And to make things worse, the list of morphological abbreviations actually has a syntax that is closer to usual collection code lists than the actual collection code list.

**Comment by gsautter** [comment]
This one is a bit of a nightmare in general ... specimen counts given as words ("two" instead of "2") are hard to parse, and life stage _plus_ gender is pretty rare as well. Will need a bit of thought, this one.
